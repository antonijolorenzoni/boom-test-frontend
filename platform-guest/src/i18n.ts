import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';
import detector from 'i18next-browser-languagedetector';

import { resources } from './translations';

i18n
  .use(detector)
  .use(initReactI18next)
  .init({
    resources,
    interpolation: {
      escapeValue: false,
    },
    fallbackLng: 'en',
    load: 'languageOnly',
    detection: {
      order: ['localStorage', 'navigator'],
      caches: ['localStorage'],
    },
  });

export default i18n;
