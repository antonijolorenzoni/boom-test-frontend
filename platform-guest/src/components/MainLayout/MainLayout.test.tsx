import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import 'jest-styled-components';

import i18n from 'i18n';

import { MainLayout } from '.';

beforeEach(() => {
  Object.defineProperty(window, 'localStorage', {
    value: {
      getItem: jest.fn(() => 'SWT15-763'),
    },
  });
});

jest.mock('axios', () => {
  const axios = jest.requireActual('axios');
  axios.create = jest.fn().mockImplementation(() => axios);

  axios.get = jest.fn().mockImplementation((url) => {
    switch (url) {
      case `/orders/SWT15-763`:
        return Promise.resolve({
          data: {
            orderCode: 'SWT15-763',
            orderType: 'REAL_ESTATE',
          },
        });
    }
  });

  return axios;
});

test('<MainLayout /> renders correctly saving a snapshot', async () => {
  const { container } = render(
    <MainLayout>
      <div>content here!</div>
    </MainLayout>
  );

  await waitFor(() => expect(screen.getByText(i18n.t('header.aboutUs') as string)).toBeInTheDocument());

  expect(container).toMatchSnapshot();
});
