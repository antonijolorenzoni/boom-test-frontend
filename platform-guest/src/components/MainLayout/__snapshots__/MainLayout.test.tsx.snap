// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainLayout /> renders correctly saving a snapshot 1`] = `
.c1 {
  padding: 35px 31px;
  top: 0;
  width: 100vw;
  position: relative;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  margin-top: 20px;
  margin-bottom: 20px;
  opacity: 0.7;
}

.c3:hover {
  opacity: 1;
}

.c3:active {
  opacity: 1;
}

.c2 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #ffffff;
}

.c4 {
  font-size: 25px;
}

.c5 {
  cursor: pointer;
}

.c0 {
  height: 100vh;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  box-sizing: border-box;
  height: calc(100vh - 234px);
}

@media screen and (max-width:576px) {
  .c6 {
    bottom: 0;
    position: fixed;
  }
}

@media screen and (max-width:768px) {
  .c6 {
    height: calc(100vh - 117px);
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <img
        alt="Boom Logo"
        height="47"
        src="logo_black.png"
      />
      <div
        style="display: flex; align-items: center;"
      >
        <a
          class="c2"
          href="https://boom.co/about-us/"
          target="_blank"
        >
          <div
            class="c3"
            style="margin-right: 42px;"
          >
            <i
              aria-hidden="true"
              class="c4 material-icons-outlined material-icons"
              role="presentation"
              size="25"
              style="margin-right: 10px;"
            >
              camera_alt
            </i>
            <h2
              style="margin: 0px; padding: 0px; font-size: 17px; font-weight: 500; line-height: 24px; color: rgb(255, 255, 255); text-align: left; text-transform: none;"
            >
              About us
            </h2>
          </div>
        </a>
        <a
          class="c2"
          data-testid="faq-link"
          href="https://boom.co/businesses/faq-business-owner-real-estate/"
          target="_blank"
        >
          <div
            class="c3"
            style="margin-right: 42px;"
          >
            <i
              aria-hidden="true"
              class="c4  material-icons"
              role="presentation"
              size="25"
              style="margin-right: 10px;"
            >
              chat_bubble_outline
            </i>
            <h2
              style="margin: 0px; padding: 0px; font-size: 17px; font-weight: 500; line-height: 24px; color: rgb(255, 255, 255); text-align: left; text-transform: none;"
            >
              FAQ
            </h2>
          </div>
        </a>
        <div
          class="c3"
          data-testid="wrapper-logout-test-id"
        >
          <i
            aria-hidden="true"
            class="c4  material-icons"
            role="presentation"
            size="25"
            style="margin-right: 10px;"
          >
            power_settings_new
          </i>
          <h2
            style="margin: 0px; padding: 0px; font-size: 17px; font-weight: 500; line-height: 24px; color: rgb(255, 255, 255); text-align: left; text-transform: none;"
          >
            Log out
          </h2>
        </div>
        <div
          style="height: 22px; margin-left: 32px; margin-right: 0px;"
        >
          <img
            alt="flag-[object Object]"
            class="c5"
            src="[object Object]"
          />
        </div>
      </div>
    </div>
    <div
      class="c6"
    >
      <div>
        content here!
      </div>
    </div>
  </div>
</div>
`;
